spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/actions/RewriteTablePathSparkAction.java

```diff
@@ -728,4 +724,22 @@ private String getMetadataLocation(Table tbl) {
         !metadataDir.isEmpty(), "Failed to get the metadata file root directory");
     return metadataDir;
   }
+
+  @VisibleForTesting
```

Comment:
Thanks for catching this bug, you probably tried but i am wondering is there no other way to force this exception except having this extra visibleForTesting method?  The code doesnt try to de-serialize it in a normal rewriteTablePath call?

Reply:
Not possible to force deserializing in Spark local cluster since the original value is cached.